lib/src/photo_view_pageview_wrapper.dart

```diff
@@ -0,0 +1,262 @@
+      }
+    }
+  }
+
+  @override
+  void onScaleStart(ScaleStartDetails detail) {
+    _startPosition = detail.focalPoint;
+    _lastScrollPixels =
+        pageViewController.position.pixels ?? _freshScrollPixels;
+    _scaleStartDetail = detail;
+  }
+
+  @override
+  void onScaleUpdate(ScaleUpdateDetails detail) {
+    if (childCallbacks.isNotEmpty) {
+      final Offset delta = detail.focalPoint - _startPosition;
+      bool pageViewShouldMove = false;
+      for (final callback in childCallbacks) {
+        final callbackIndex = callbackIntMap[callback];
+        if (callbackIndex != null && callbackIndex == _currentSelectPage) {
+          if (needNotifyChildEnd) {
+            callback.onScaleUpdate(detail);
+          } else {
+            final bool childCanMove = callback.canMove(detail.scale, delta);
+            if (childCanMove != null && childCanMove) {
+              needNotifyChildEnd = true;
+              callbackStartMap[callback] = childCanMove;
+              callback.onScaleStart(_scaleStartDetail);
+            } else {
+              pageViewShouldMove = true;
+            }
+          }
+        }
+      }
+
+      if (pageViewShouldMove) {
+        movePageViewSelf(detail);
+      }
+    } else {
+      movePageViewSelf(detail);
+    }
+  }
+
+  //scroll to next or pre page after finger up
+  void animateToPage(ScrollPositionWithSingleContext scrollPosition) {
+    final tail = pageViewController.page - pageViewController.page.floor();
+    if (_lastDelta == null) {
+      return;
+    }
+
+    if (_lastDelta.dx > 0) {
+      //finger move right
+      if (tail >= 0.5) {
+        pageViewController.animateToPage(pageViewController.page.ceil(),
+            curve: Curves.decelerate,
+            duration: Duration(milliseconds: durationMs));
+      } else {
+        pageViewController
+          ..animateToPage(pageViewController.page.floor(),
+              curve: Curves.decelerate,
+              duration: Duration(milliseconds: durationMs));
+      }
+    } else {
+      //finger move left
+      if (tail >= 0.5) {
+        pageViewController
+          ..animateToPage(pageViewController.page.ceil(),
+              curve: Curves.decelerate,
+              duration: Duration(milliseconds: durationMs));
+      } else {
+        pageViewController
+          ..animateToPage(pageViewController.page.floor(),
+              curve: Curves.decelerate,
+              duration: Duration(milliseconds: durationMs));
+      }
+    }
+  }
+
+  void onScrollUpdate(double pixels) {
+    _freshScrollPixels = pixels;
+  }
+
+  void movePageViewSelf(ScaleUpdateDetails detail) {
+    final ScrollPositionWithSingleContext scrollPosition =
+        pageViewController.position;
+    final Offset delta = detail.focalPoint - _startPosition;
+    final value = _lastScrollPixels - delta.dx;
+    _lastDelta = delta;
+    scrollPosition.jumpToWithoutSettling(
+        value.clamp(-0.0, scrollPosition.maxScrollExtent));
```

Comment:
You can use trailing commas here.
```suggestion
        value.clamp(-0.0, scrollPosition.maxScrollExtent,),);
```

After doing this, if you run format, the code will be more readable.